### DISCUSSION TITLE
Can't install Voodoo 3 drivers for Windows 98

### DISCUSSION THREAD

**Discussion by Vort** [bochs-emu/Bochs#180] [Q&A]
I tried to install different Voodoo 3 drivers for Windows 98 from this page:
http://falconfly.3dfx.pl/voodoo3.htm

No matter what methods I used, result was the same: device does not work.
Because failure happens at early stage, I began to suspect problems with PCI.

After comparing device manager lists with VirtualBox, I found that in Bochs there are no "PCI bus" device:
![image](https://github.com/bochs-emu/Bochs/assets/1242858/4461bdd3-3c62-4108-8d83-bae87985e322)

If absence of "PCI bus" in Windows 98 is a bug, please create issue about it.
If it happens because of my mistake, please explain what I'm doing wrong.

**Comment by Vort** [comment]
Also no "PCI bus" appear for Windows 95:

![bochs_win95](https://github.com/bochs-emu/Bochs/assets/1242858/527d4148-f092-4328-a485-4922346418af)

**Comment by Vort** [comment]
I figured out that I can manually add "PCI bus" device:
![bochs_pci_install](https://github.com/bochs-emu/Bochs/assets/1242858/046a6fe5-6430-45cc-9254-990f3ad980ff)

After that, Voodoo 3 driver (1.07.00) installs fine:
![bochs_voodoo3_win98](https://github.com/bochs-emu/Bochs/assets/1242858/91e42f89-67d5-458e-ab26-a3a3d25dbc70)

But it is better to find out why automatic PCI bus installation is not started.

**Reply by sskras** [reply]
> But it is better to find out why automatic PCI bus installation is not started.

Comment from a passer-by: 100% agreed.
I wonder if PCI bus isn't automatically recognized in [86Box](https://github.com/86Box/86Box) too...

**Reply by vruppert** [reply]
I know about the missing PCI bus after installing Win9x. Since the Bochs BIOS (as well as SeaBIOS) has only a stub for the PnP BIOS, it is not possible to get the list of installed devices.

**Reply by sskras** [reply]
Thanks for bringing clarity!
